libyul/backends/evm/EVMCodeTransform.cpp

```diff
@@ -642,9 +642,24 @@ void CodeTransform::operator()(ForLoop const& _forLoop)
 	m_assembly.appendLabel(loopEnd);
 
 	finalizeBlock(_forLoop.pre, stackStartHeight);
+	m_context->forLoopStack.pop();
 	m_scope = originalScope;
 }
 
+void CodeTransform::operator()(Break const& _break)
+{
+	yulAssert(!m_context->forLoopStack.empty(), "Invalid break-statement. Requires surrounding for-loop in code generation.");
+	m_assembly.setSourceLocation(_break.location);
+	m_assembly.appendLabel(m_context->forLoopStack.top().done);
```

Comment:
Should this skip the post part?

Reply:
AFAIU a break always skips post. I may be wrong wrt. Yul/Solidity. any thoughts, @chriseth?

Reply:
Yes, break directly exists the loop.